src/Illuminate/Contracts/Auth/MustVerifyEmail.php

```diff
@@ -31,4 +33,12 @@ public function sendEmailVerificationNotification();
      * @return string
      */
     public function getEmailForVerification();
+
+    /**
+     * Scope a query to only include users with verified email.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeHasVerifiedEmail(Builder $query);
```

Comment:
This is a breaking change for any class that implements this interface.

Reply:
I also don't think this needs to be added to the interface since this is leaking Eloquent into the auth component.

Reply:
Yes, I was hesitating about that. I remove that from the interface.